### ISSUE TITLE
Copy from built-in documentation with selection starts from line start sometimes copy the upper line

### ISSUE THREAD

**Issue by jsjtxietian** [godotengine/godot#87467]
### Tested versions

Godot v4.3.dev (3c596094a)

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 31.0.15.3619) - 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (16 Threads)

### Issue description

When debugging https://github.com/godotengine/godot/issues/87463, I did some test myself and found a new issue, keep it seperate because it may have different underlying reasons.

I want to copy `If exact_match is false`

But instead I got 
```
This is useful for code that needs to run only once when an action is pressed, instead of every frame while it's pressed.

If exact_match is false
```
In my clipboard

See the video:

https://github.com/godotengine/godot/assets/8315986/f3ed4b91-b7a0-42c7-80a1-aa1e489cae8b

### Steps to reproduce

1. Open built-in doc
2. Copy from line start, and paste and observe.

### Minimal reproduction project (MRP)

N/A

**Comment by dalexeev** [comment]
Most likely this is a `RichTextLabel` bug. CC @bruvzg
